### PR TITLE
Tele COPE

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
What's worse than a SOM engineer throwing a COPE sentry at you?
A SOM SL calling in a teleporter delivered COPE on you.

Adds the ability to summon up to 2 COPE sentries via teleporter. Like TGMC sentry/supply pods, this is enabled simply by activating the teleporter asset, and lasing like all air support.

For maints: I also reorganised the files, as it was all crammed in one giant DM. Only the stuff in the teleport.dm file is new though.
## Why It's Good For The Game
While the teleporter is situationally very strong, in practical terms drop pods are SIGNIFICANTLY more useful, due to being very flexible for individual drops.

This adds a bit more utility to the teleporter, even when you're not using it to deploy troops or supplies.
## Changelog
:cl:
add: Campaign: Activating the teleporter asset gives access to a teleport COPE air support type
/:cl:

### CODE REVIEW

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/fire_support/fire_support.dm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		playsound(target_turf, start_sound, 100)
+
+///Selects the final target turf(s) and calls impact procs
+/datum/fire_support/proc/select_target(turf/target_turf)
+	var/list/turf_list = list()
+	for(var/turf/spread_turf in RANGE_TURFS(scatter_range, target_turf))
+		turf_list += spread_turf
+	for(var/i = 1 to impact_quantity)
+		var/turf/impact_turf = pick(turf_list)
+		addtimer(CALLBACK(src, PROC_REF(do_impact), impact_turf), 0.15 SECONDS * i)
```
[the remark this answers is not in the document]
Comment (Lumipharon) [reply code/datums/fire_support/fire_support.dm:85]: I unno, old code or something, probably from CAS GAU

File: code/game/objects/structures/orbital_cannon.dm [hunk 2]
```diff
@@ -399,9 +399,7 @@
 	set waitfor = FALSE
 	. = ..()
 	var/range_num = max(9 - inaccuracy_amt, 6)
-	var/list/turf_list = list()
-	for(var/turf/T in range(range_num, target))
-		turf_list += T
+	var/list/turf_list = range(range_num, target)
```
Comment (TiviPlus) [inline code/game/objects/structures/orbital_cannon.dm:402]: RANGE_TURFS, range() returns all atoms

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest looks ok

File: code/game/objects/structures/orbital_cannon.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/structures/orbital_cannon.dm:402]: whoops

**Pull request merged**